examples/example.rs

```diff
@@ -43,7 +45,9 @@ fn main() {
     }
 
     // middleware is optional and can be registered with `utilize`
-    server.utilize(logger);
+    // issue #15444
+    let logger_handler: fn(&Request, &mut Response) -> MiddlewareResult = logger;
+    server.utilize(logger_handler);
```

Comment:
So I guess we leave it as is and revisit later on when the variants of https://github.com/rust-lang/rust/issues/20178 are done?